homeassistant/components/ecobee/fan.py

```diff
@@ -0,0 +1,161 @@
+
+    platform = entity_platform.async_get_current_platform()
+
+    platform.async_register_entity_service(
+        SERVICE_SET_VENTILATOR_MIN_ON_TIME_HOME,
+        {
+            vol.Required(ATTR_VENTILATOR_MIN_ON_TIME_HOME): vol.Coerce(int),
+        },
+        "set_ventilator_min_on_time_home",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_VENTILATOR_MIN_ON_TIME_AWAY,
+        {
+            vol.Required(ATTR_VENTILATOR_MIN_ON_TIME_AWAY): vol.Coerce(int),
+        },
+        "set_ventilator_min_on_time_away",
+    )
+
+    platform.async_register_entity_service(
+        SERVICE_SET_VENTILATOR_TIMER,
+        {
+            vol.Required(ATTR_IS_VENTILATOR_TIMER_ON): cv.boolean,
+        },
+        "set_ventilator_timer",
+    )
+
+
+class EcobeeVentilator(FanEntity):
+    """A ventilator class for an ecobee thermostat with ventilator attached."""
+
+    def __init__(self, data, thermostat_index):
+        """Initialize ecobee ventilator platform."""
+        self.data = data
+        self.thermostat_index = thermostat_index
+        self.thermostat = self.data.ecobee.get_thermostat(self.thermostat_index)
+        ecobee_name = self.thermostat["name"]
+        self._attr_name = f"{ecobee_name} Ventilator"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.thermostat["identifier"])},
+            manufacturer=MANUFACTURER,
+            model=ECOBEE_MODEL_TO_NAME.get(self.thermostat["modelNumber"]),
+            name=self.name,
+        )
+        self._attr_unique_id = f"{self.thermostat['identifier']}"
+
+        self.update_without_throttle = False
+
+    @property
+    def available(self):
+        """Return if device is available."""
+        return self.thermostat["runtime"]["connected"]
+
+    async def async_update(self):
+        """Get the latest state from the thermostat."""
+        await self.data.update()
+        self.thermostat = self.data.ecobee.get_thermostat(self.thermostat_index)
+
+    @property
+    def is_on(self):
+        """Return True if the ventilator is on."""
+        return "ventilator" in self.thermostat["equipmentStatus"]
+
+    @property
+    def extra_state_attributes(self):
+        """Return device specific state attributes."""
+        return {
+            "ventilator type": self.thermostat["settings"]["ventilatorType"],
+            "ventilator_min_on_time_home": self.thermostat["settings"][
+                "ventilatorMinOnTimeHome"
+            ],
+            "ventilator_min_on_time_away": self.thermostat["settings"][
+                "ventilatorMinOnTimeAway"
+            ],
+            "is_ventilator_timer_on": self.thermostat["settings"][
+                "isVentilatorTimerOn"
+            ],
+        }
+
+    def turn_on(
+        self,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Turn on the ventilator 20 min manual timer."""
```

Comment:
Shouldn't these be manual e.g. no timer?

Reply:
This type of ventilator does not offer manual on/off only via timer and time per minutes on home or away